[mod] source code url: GIT_URL + '/tree/' + GIT_BRANCH

return42 opened the pull request:
The link 'Source code' in the footer should link to the branch URL:

    GIT_URL + '/tree/' + GIT_BRANCH

This URL fits at least for github and gitlab.

BTW clear the naming and replace:

- `searx_git_url` and `searx_git_branch` by `GIT_URL` and `GIT_BRANCH`
- `searx_version` by `searxng_version`

----

To test create a clone of this PR:

```
$ cd /tmp
$ git clone https://github.com/return42/searxng/
$ cd searxng/
$ git checkout git-url-branch
$ make buildenv
$ git diff
```

```diff
diff --git a/utils/brand.env b/utils/brand.env
index 31afce53..6fd66ebe 100644
--- a/utils/brand.env
+++ b/utils/brand.env
@@ -1,5 +1,5 @@
 export SEARXNG_URL=''
 export SEARXNG_PORT='8888'
 export SEARXNG_BIND_ADDRESS='127.0.0.1'
-export GIT_URL='https://github.com/searxng/searxng'
-export GIT_BRANCH='master'
+export GIT_URL='https://github.com/return42/searxng/'
+export GIT_BRANCH='git-url-branch'
```

```
make run
```

The source code link in the footer should link to this branch https://github.com/return42/searxng/tree/git-url-branch

return42 committed:
[mod] source code url: GIT_URL + '/tree/' + GIT_BRANCH

The link 'Source code' in the footer should link to the branch URL:

    GIT_URL + '/tree/' + GIT_BRANCH

This URL fits at least for github and gitlab.

BTW clear the naming and replace:

- `searx_git_url` and `searx_git_branch` by `GIT_URL` and `GIT_BRANCH`
- 'searx_version' by 'searxng_version'

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/webapp.py:
```diff
@@ -1273,7 +1274,8 @@ def get_key(engine_stat):
         selected_engine_name = selected_engine_name,
-        searx_git_branch = GIT_BRANCH,
+        GIT_URL = GIT_URL,
+        GIT_BRANCH = GIT_BRANCH,
```
I haven't tested, but it seems these two variables are defined in the `render`  function, so there are not needed here ?

return42 replied:
> I haven't tested, but it seems these two variables are defined in the `render`  function, so there are not needed here ?

Oops .. thanks for pointing out .. [fixed](https://github.com/searxng/searxng/compare/7b43d96fc69ab0e273bbfd9c842b4c6239eb22c0..e34782e0b9b1776b95ed418bfc850a552ccd2909)

return42 committed:
[mod] source code url: add 'general.git_url_format' to settings.yml

Suggested-by: @dalf https://github.com/searxng/searxng/pull/1295#issuecomment-1152608817
Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
